lib/Kconfig.debug

```diff
@@ -2582,6 +2582,15 @@ config FORTIFY_KUNIT_TEST
 	  by the str*() and mem*() family of functions. For testing runtime
 	  traps of FORTIFY_SOURCE, see LKDTM's "FORTIFY_*" tests.
 
+config LONGEST_SYM_KUNIT_TEST
+	tristate "Test the longest symbol possible" if !KUNIT_ALL_TESTS
```

Comment:
This doesn't built as a module:
```
ERROR: modpost: missing MODULE_LICENSE() in lib/longest_symbol_kunit.o
ERROR: modpost: "kallsyms_lookup_name" [lib/longest_symbol_kunit.ko] undefined!
```

The easy thing to do would be to make this a ``bool`` option, but this is exactly the sort of thing we probably should test as a module, too (maybe that's why the longest+1 name is working: the compiler/linker are fine, but the module tools (e.g. modpost) aren't). Fixing it up to work as a module would lose us some of the simplicity of the KUnit port, though.

Reply:
Interesting ... yes, that may be very well the reason for working. That is good because I was feeling quite uneasy about it.  

I will look into it and make it a module. 

Thanks for pointing it out :) :+1:

Reply:
@sulix I have updated the code, so now it actually compiles and can be executed as a module (as referenced here: https://kunit.dev/release_notes.html?highlight=module ).

Incredible as it sounds, the test with the longest symbol +1, when executed as a module, passes. Is not clear to me why. I have to delve into this further.

```
sergio@ubuntu:/lib/modules/6.1.0-rc1-188160-g3ee25d98b592/kernel/lib$ dmesg -HW &
[1] 3207
sergio@ubuntu:/lib/modules/6.1.0-rc1-188160-g3ee25d98b592/kernel/lib$ sudo modprobe longest_symbol_kunit
[feb27 21:49]     # Subtest: longest-symbol
[  +0,000002]     1..4
[  +0,000075]     ok 1 - test_longest_symbol
[  +0,028325]     ok 2 - test_longest_symbol_kallsyms
[  +0,000237]     ok 3 - test_longest_symbol_plus1
[  +0,012761]     ok 4 - test_longest_symbol_plus1_kallsyms
[  +0,000005] # longest-symbol: pass:4 fail:0 skip:0 total:4
[  +0,000001] # Totals: pass:4 fail:0 skip:0 total:4
[  +0,000001] ok 1 - longest-symbol
```

Reply:
I have taken a look: the test passes unexpectedly because the symbol is simply `LONGEST_SYM_PLUS1`, not the expected one (very long).

This is due to two issues:

  - A typo in `LONGEST_SYM_PLUS1` vs. `LONGEST_SYM_NAME_PLUS1`.

  - The preprocessor will concatenate before expanding. So you need one level of indirection more, e.g. with the `__PASTE` macro.

```diff
diff --git a/lib/longest_symbol_kunit.c b/lib/longest_symbol_kunit.c
index 575edca77864..10d2e9967647 100644
--- a/lib/longest_symbol_kunit.c
+++ b/lib/longest_symbol_kunit.c
@@ -18,13 +18,13 @@
 #define DDDDI(name) DDDI(n##name##name)
 #define DDDDDI(name) DDDDI(n##name##name)
 
-#define PLUS1(name) name##e
+#define PLUS1(name) __PASTE(name,e)
 
 /*Generate a symbol whose name length is 511 */
 #define LONGEST_SYM_NAME  DDDDDI(g1h2i3j4k5l6m7n)
 
 /*Generate a symbol whose name length is 512 */
-#define LONGEST_SYM_PLUS1 PLUS1(LONGEST_SYM_NAME)
+#define LONGEST_SYM_NAME_PLUS1 PLUS1(LONGEST_SYM_NAME)
 
 int noinline LONGEST_SYM_NAME(void)
 {
```

With this:

```sh
$ nm lib/longest_symbol_kunit.o | awk 'length($3) > 100 { print length($3), $3 }'
511 snnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7n
512 snnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nnng1h2i3j4k5l6m7ng1h2i3j4k5l6m7nng1h2i3j4k5l6m7ng1h2i3j4k5l6m7ne
```

You can play with it at https://godbolt.org/z/9ach7Wq63.

Hope that helps!

Reply:
:) Great! thanks a lot for having a look and pointing out the issue! I will fix this :+1:

Reply:
My pleasure, and thanks for the delay!